src/js/sources-util.js

```diff
@@ -20,9 +20,55 @@ export function removeDuplicateMessages(msgs, sourceLatency=10) {
   return msgs.filter(Boolean).filter(isUnique);
 }
 
+/**
+ * Find authors that send messages at a specified frequency.
+ *
+ * @param {Message[]} msgs
+ * @param {Number} amountOfMsgs amount of msgs in the spam frequency
+ * @param {Seconds} time the time between the first spam message and the latest
+ * @return {String[]} authorIds
+ */
+export function getSpamAuthors(msgs, amountOfMsgs, time) {
+  const authors = [];
+
+  index(msgs).by('authorId').forEach((messages, author) => {
+    if (containsSpam(messages, amountOfMsgs, time)) {
+      authors.push(author);
+    }
+  });
+
+  return authors;
+}
+
+/** @type {(msgs: Message[], amount: Number, time: Seconds) => Boolean} */
+const containsSpam = (msgs, amount, time) => {
+  let beg = 0;
+  const timestamps = msgs.map(messageTime);
+  for (const [i, timestamp] of timestamps.entries()) {
+    // advance beg pointer until it is time seconds behind timestamp
+    while (timestamp - timestamps[beg] > time) beg++;
+    if (i - beg + 1 >= amount) return true;
+  }
```

Comment:
I might just be dumb and need an explanation of how this loop works, but I think line 49 should be a `>=` to include messages that are exactly at the set interval seconds to be counted as spam.

Reply:
No, it should be ``>`` because the end condition that we want is ``timestamp - timestamps[beg] <= time`` and ``<=`` is the opposite of ``>`` so when this loop finishes, the end condition is ``true``.